src/scripts/main.js

```diff
@@ -1,5 +1,39 @@
 'use strict';
 
+const spider = document.querySelector('.spider');
+const spiderX = spider.clientWidth;
+const spiderY = spider.clientHeight;
+
+const wall = document.querySelector('.wall');
+const wallX = wall.clientWidth;
+const wallY = wall.clientHeight;
+
 document.addEventListener('click', e => {
-  // write code here
+  if (e.target.tagName === 'BODY') {
+    return;
+  }
+
+  let x = e.offsetX;
+  let y = e.offsetY;
+
+  switch (true) {
+    case (x < 0 + spiderX / 2) :
+      x = 0 + spiderX / 2;
+      break;
+    case (x > wallX - spiderX) :
+      x = wallX - spiderX / 2;
+      break;
+  }
+
+  switch (true) {
+    case (y < 0 + spiderY / 2) :
+      y = 0 + spiderY / 2;
+      break;
+    case (y > wallY - spiderY) :
+      y = wallY - spiderY / 2;
+      break;
+  }
```

Comment:
You repeating yourself two times here. You could create a function that spiderX/spiderY and wallX/wallY as arguments and return x or y. It would reduce the code and you won't repeat yourself